products/compute/api.yaml

```diff
@@ -7553,6 +7553,172 @@ objects:
+      guides:
+        'Official Documentation': 'https://cloud.google.com/load-balancing/docs/negs/serverless-neg-concepts'
+      api: 'https://cloud.google.com/compute/docs/reference/rest/beta/regionNetworkEndpointGroups'
+    description: |
+      A regional NEG that can support Serverless Products.
+    async: !ruby/object:Api::OpAsync
+      operation: !ruby/object:Api::OpAsync::Operation
+        kind: 'compute#operation'
+        path: 'name'
+        base_url: 'projects/{{project}}/regions/{{region}}/operations/{{op_id}}'
+        wait_ms: 1000
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'targetLink'
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'status'
+        complete: 'DONE'
+        allowed:
+          - 'PENDING'
+          - 'RUNNING'
+          - 'DONE'
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error/errors'
+        message: 'message'
+    parameters:
+      - !ruby/object:Api::Type::ResourceRef
+        name: 'region'
+        resource: 'Region'
+        imports: 'name'
+        description: |
+          A reference to the region where the Serverless NEGs Reside.
+        required: true
+    properties:
+      - !ruby/object:Api::Type::Integer
+        name: 'id'
+        description: 'The unique identifier for the resource.'
+        output: true
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          Name of the resource; provided by the client when the resource is
+          created. The name must be 1-63 characters long, and comply with
+          RFC1035. Specifically, the name must be 1-63 characters long and match
+          the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?` which means the
+          first character must be a lowercase letter, and all following
+          characters must be a dash, lowercase letter, or digit, except the last
+          character, which cannot be a dash.
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: |
+          An optional description of this resource. Provide this property when
+          you create the resource.
+      - !ruby/object:Api::Type::Enum
+        name: 'networkEndpointType'
+        required: true
+        description: |
+          Type of network endpoints in this network endpoint group.
+        values:
+          - :INTERNET_IP_PORT
+          - :INTERNET_FQDN_PORT
+          - :SERVERLESS
+          - :NON_GCP_PRIVATE_IP_PORT
+          - :GCE_VM_IP_PORT
+      - !ruby/object:Api::Type::NestedObject
+        name: 'cloudRun'
+        exactly_one_of:
+          - cloudRun
+          - cloudFunction
+          - appEngine
+        description: |
+          Only valid when networkEndpointType is "SERVERLESS".
+          Only one of cloud_run, app_engine or cloud_function may be set.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'service'
```

Comment:
Is this field required?

Reply:
yeah it is

Reply:
Actually, i'm going to go with not required, the google api docs are not clear and i rather let terraform return the error back in incorrect configurations. The CLI makes it optional i think.

https://cloud.google.com/sdk/gcloud/reference/beta/compute/network-endpoint-groups/create#--cloud-function-name